server/src/main/java/org/apache/iotdb/db/metadata/MTree.java

```diff
@@ -215,6 +215,19 @@ public void setStorageGroup(String path) throws PathErrorException {
     setDataFileName(path, cur);
   }
 
+  public void deleteStorageGroup(String path) throws PathErrorException {
+    if (!checkStorageGroup(path)) {
+      throw new PathErrorException(String.format("The path %s is not a deletable storage group", path));
+    }
+    MNode cur = getNodeByPath(path);
```

Comment:
Instead of using `checkStorageGroup`, I think it is better to use
```
if (!cur.isStorageGroup()) {
    throw ...
}
```
which could reduce one seek in the MTree.